xwiki-platform-core/xwiki-platform-oldcore/src/main/java/org/xwiki/store/hibernate/AbstractHibernateAdapter.java

```diff
@@ -0,0 +1,246 @@
+    {
+        return getTableName(persistentClass.getTable());
+    }
+
+    @Override
+    public String getTableName(Table table)
+    {
+        return getTableName(table.getName());
+    }
+
+    @Override
+    public String getTableName(String tableName)
+    {
+        return tableName;
+    }
+
+    @Override
+    public boolean isCompressed(PersistentClass entity)
+    {
+        return entity.getMetaAttribute(META_ATTRIBUTE_COMPRESSED) != null;
+    }
+
+    @Override
+    public String escapeDatabaseName(String databaseName)
+    {
+        String closeQuote = String.valueOf(getDialect().closeQuote());
+        return getDialect().openQuote() + databaseName.replace(closeQuote, closeQuote + closeQuote) + closeQuote;
+    }
+
+    // Global
+
+    @Override
+    public SessionFactory getSessionFactory()
+    {
+        return this.hibernateStore.getSessionFactory();
+    }
+
+    @Override
+    public Dialect getDialect()
+    {
+        return this.hibernateStore.getDialect();
+    }
+
+    // Database operations
+
+    @Override
+    public void updateDatabase(Metadata metadata) throws HibernateStoreException
+    {
+        try {
+            SessionFactory sessionFactory = getSessionFactory();
+
+            // Custom update before the Hibernate standard DDL
+            try (Session session = sessionFactory.openSession()) {
+                updateDatabaseBegin(metadata, session);
+            }
+
+            // Hibernate standard DDL
+            updateDatabaseStandard(metadata);
+
+            // Custom update after the Hibernate standard DDL
+            try (Session session = sessionFactory.openSession()) {
+                updateDatabaseAfter(metadata, session);
+            }
+        } catch (Exception e) {
+            throw new HibernateException("Failed to update the database", e);
+        }
+    }
+
+    private void updateDatabaseStandard(Metadata metadata) throws HibernateStoreException
+    {
+        SchemaUpdate updater = new SchemaUpdate();
+        updater.execute(EnumSet.of(TargetType.DATABASE), metadata);
+        List<Exception> exceptions = updater.getExceptions();
+
+        if (!exceptions.isEmpty()) {
+            // Print the errors
+            for (Exception exception : exceptions) {
+                this.logger.error(exception.getMessage(), exception);
+            }
+
+            throw new HibernateStoreException("Failed to update the database. See the previous log for all errors",
+                exceptions.get(0));
+        }
+    }
+
+    /**
+     * Execute everything not handled by the Hibernate DDL which needs to be done before.
+     * 
+     * @param metadata the configuration of the database to apply
+     * @param session the session in which to execute the update
+     * @throws HibernateStoreException when failing to update the database
+     */
+    protected void updateDatabaseBegin(Metadata metadata, Session session) throws HibernateStoreException
```

Comment:
Is it worth being documented with explanations of when to override as?
1. the default implementation is empty
1. no sub-class overrides it

Reply:
I don't have much real life example in mind, it's here mostly in case something could need to be done before the standard Hibernate update is executed (some limitation, bug, etc.). I preferred having something to not be stuck.